### PR TITLE
feat: add chart interactions on click / dblclick

### LINKED ISSUE #1504: Double click chart to interact
Charts are clunky at the moment and leave much to be desired on interactivity. Double click to interact is a good first start.

### PR DESCRIPTION
**Pull request by AyushAgrawal-A2**
closes #1504

- [x] Hover - Cursor pointer
- [x] Click - Selects corresponding code cell and move chart to top in case of overlapping charts
- [x] Meta / Ctrl + Click - Select code cell + Move to top + Add to Multicursor selection
- [x] DoubleClick - Opens code editor for corresponding code cell

Resizing functionality is same as before.

Built on top of #1494 to reuse `openCodeEditor` function extracted in that pr.

### CODE REVIEW

File: quadratic-client/src/app/gridGL/HTMLGrid/hoverCell/HoverCell.tsx [hunk 1]
```diff
@@ -189,6 +189,7 @@ export const HoverCell = () => {
         position: 'absolute',
         visibility: cell ? 'visible' : 'hidden',
         pointerEvents: 'none',
+        zIndex: 999999,
```
Comment (davidfig) [inline quadratic-client/src/app/gridGL/HTMLGrid/hoverCell/HoverCell.tsx:192]: Let's try to keep zIndex as small as possible to get the behavior we want. There may be something that needs to go on top of this.

Comment (AyushAgrawal-A2) [reply quadratic-client/src/app/gridGL/HTMLGrid/hoverCell/HoverCell.tsx:192]: @davidfig 
I needed an arbitrarily large number, when moving a chart on top I am increasing its zIndex to 1 + current max. So zIndex of charts can keep increasing. All charts have 0 zIndex on sheet open.
I think hover popups should always be on top of everything, and zIndex max can go upto int32 max value so there is room for placing something above this.
Still practically even 100 should be enough. It this ok ?

Comment (davidfig) [reply quadratic-client/src/app/gridGL/HTMLGrid/hoverCell/HoverCell.tsx:192]: Ah. So charts are always z+1. It makes sense why you picked a large number. Maybe 1000 to ensure we don't run into a problem here.

Comment (AyushAgrawal-A2) [reply quadratic-client/src/app/gridGL/HTMLGrid/hoverCell/HoverCell.tsx]: @davidfig 
Done

**Review by davidfig — APPROVED** [review]

**Comment by davidkircos** [comment]
@AyushAgrawal-A2 can you move this to a new branch on `quadratic` so we get a preview branch for it?

**Comment by AyushAgrawal-A2** [comment]
move to quadratic repo from fork #1625

**Pull request closed without merging**